src/main/java/dk/kb/image/kaltura/DsKalturaClient.java

```diff
@@ -60,34 +67,36 @@ public DsKalturaClient(String kalturaUrl, String userId, int partnerId, String a
      * @return The Kaltura id (internal id). Return null if the refId is not found.
      * @throws IOException if Kaltura called failed, or more than 1 entry was found with the referenceId.
      */
+    @SuppressWarnings("unchecked")
     public String getKulturaInternalId(String referenceId) throws IOException{
-
-        KalturaFilterPager pager = new KalturaFilterPager();
-        pager.pageSize = 10;
+        
+        System.out.println("calling with referenceId:"+referenceId);
+        MediaEntryFilter filter = new MediaEntryFilter();
+        filter.setReferenceIdEqual(referenceId);
+        //filter.idEqual("0_g9ys622b"); //Example to search for id
                 
-        KalturaMediaEntryFilter filter = new KalturaMediaEntryFilter();                
-        //filter.searchTextMatchAnd="\""+referenceId+"\""; // Can also find with freetext search (quotes), but this can give false matches.
-        filter.referenceIdEqual=referenceId;        
-        KalturaMediaListResponse list=null;
-        try {
-          list = client.getMediaService().list(filter,pager);               
-        }
-        catch(KalturaApiException e) {            
-           throw new IOException("Error searching kaltura for referenceId:"+referenceId);
-        }
+        FilterPager pager = new FilterPager();
+        pager.setPageIndex(10);
+        
+        ListMediaBuilder request =  MediaService.list(filter);               
+                                
+        //Getting this line correct was very hard. Little documentation and has to know which object to cast to.                
+        Response <ListResponse<MediaEntry>> response = (Response <ListResponse<MediaEntry>>) APIOkRequestsExecutor.getExecutor().execute(request.build(client));
+        List<MediaEntry> mediaEntries = response.results.getObjects();           
```

Comment:
Yes, It seems quite hard as well! Could you maybe provide a few words as comments on how you got there or what you think is happening in the ` APIOkRequestsExecutor`